streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java

```diff
@@ -49,6 +49,9 @@
 /**
  * {@code StreamsBuilder} provide the high-level Kafka Streams DSL to specify a Kafka Streams topology.
  *
+ * It is a requirement that the processing logic (Topology) be defined in a deterministic way.
```

Comment:
Nit: insert `<p>` tag to actually get the new paragraph rendered.

Nit: `Topology -> `{@link Topology}`

It's not really clear what "deterministic" means. We should elaborate more.